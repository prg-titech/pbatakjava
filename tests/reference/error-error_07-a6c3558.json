{
    "basename": "error-error_07-a6c3558",
    "cmd": "{infile}",
    "infile": "tests/error_07/build.sh",
    "infile_hash": "a82c1b8cda61005b283ddf820907ff015533e0e672f4656ca33f0ccb",
    "stdout": "error-error_07-a6c3558.stdout",
    "stdout_hash": "0885bb880120972033565c04e1b8d918879bab7fbf0089d1a971c3d6",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}