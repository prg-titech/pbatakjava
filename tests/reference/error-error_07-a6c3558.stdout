
public interface Point {
}

public interface Point_Factory {
  public Point make(int x, int y);
}

public class Point_v1 implements Point {
  public int x;

  public int y;

  public Point_v1(int x, int y) {
    this.x = x;
    this.y = y;
  }
}

public class Point_v1_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v1(x, y);
  }
}

public class Point_v2 implements Point {
  public int x;

  public int y;

  public Point_v2(int x, int y) {
    this.x = x;
    this.y = y;
  }
}

public class Point_v2_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v2(x, y);
  }
}

public class Point_v3 implements Point {
  public int x;

  public int y;

  public Point_v3(int x, int y) {
    this.x = x;
    this.y = y;
  }
}

public class Point_v3_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v3(x, y);
  }
}

public interface Line {
}

public interface Line_Factory {
  public Line make(Point_Factory V, Point_Factory W, Point p1, Point p2);
}

public class Line_v1 implements Line {
  public Point_Factory V;

  public Point_Factory W;

  public Point p1;

  public Point p2;

  public Line_v1(Point_Factory V, Point_Factory W, Point p1, Point p2) {
    this.V = V;
    this.W = W;
    this.p1 = p1;
    this.p2 = p2;
  }
}

public class Line_v1_Factory implements Line_Factory {
  public Line make(Point_Factory V, Point_Factory W, Point p1, Point p2) {
    return new Line_v1(V, W, p1, p2);
  }
}

public class Test {
  public static void main(String[] args) {
    Line l1 = new Line_v1(new Point_v2_Factory(), new Point_v2_Factory(), new Point_v2(3, 2), new Point_v2(2, 2));
  }
}
